app/helpers/vaccination_records_helper.rb

```diff
@@ -27,4 +27,70 @@ def vaccination_record_status_tag(vaccination_record)
 
     govuk_tag(text:, colour:)
   end
+
+  def vaccination_record_sync_status_tag(vaccination_record)
+    status = vaccination_record.sync_status
+    text = VaccinationRecord.human_enum_name(:sync_statuses, status)
+
+    colour =
+      case status
+      when :synced
+        "green"
+      when :pending
+        "blue"
+      when :failed
+        "red"
+      when :cannot_sync
+        "orange"
+      else
+        "grey"
+      end
+
+    govuk_tag(text:, colour:)
+  end
+
+  def format_sync_status_with_info(vaccination_record)
```

Comment:
This feels like it has quite a lot of logic and complexity to be a helper method to me. I would suggest it might be better placed as its own component, and then because it's a class we can have methods for all the various parts to this.